core/primitives/src/block_header.rs

```diff
@@ -406,6 +464,35 @@ pub struct BlockHeaderV4 {
     pub hash: CryptoHash,
 }
 
+/// V4 -> V5: Add permanent_contracts_root to inner_rest
```

Comment:
It's actually called `global_contract_root`

Also it seems like those two names are used interchangably in the code, I would suggest picking one and sticking to it. My favourite is global. 

Also ditto for singular and plural forms (contract vs contracts).